src/reducers/versions.spec.tsx

```diff
@@ -117,12 +119,84 @@ describe(__filename, () => {
 
       expect(state).toEqual({
         ...initialState,
+        expandedPaths: getParentFolders(version.file.selected_file),
         versionInfo: {
           [version.id]: createInternalVersion(version),
         },
       });
     });
 
+    it('prunes expandedPaths when load version info', () => {
```

Comment:
This test duplicates the logic of `getPrunedExpandedPaths()` which isn't necessary because you've passed on the responsibility of that logic to the function. Adding duplicated tests slows down the test suite over time and also adds a maintenance burden because any change in `getPrunedExpandedPaths()` could cause you to update unrelated tests.

Instead, I'd like to see this test inject a stub, like this:

```diff
diff --git a/src/reducers/versions.tsx b/src/reducers/versions.tsx
index 616f66d..4ef6128 100644
--- a/src/reducers/versions.tsx
+++ b/src/reducers/versions.tsx
@@ -1154,6 +1154,7 @@ export const goToRelativeDiff = ({
 const reducer: Reducer<VersionsState, ActionType<typeof actions>> = (
   state = initialState,
   action,
+  { _getPrunedExpandedPaths = getPrunedExpandedPaths } = {},
 ): VersionsState => {
   switch (action.type) {
     case getType(actions.beginFetchVersion): {
@@ -1173,7 +1174,7 @@ const reducer: Reducer<VersionsState, ActionType<typeof actions>> = (
       const { expandedPaths: expandedPathsInState } = state;
 
       const expandedPaths = expandedPathsInState
-        ? getPrunedExpandedPaths(expandedPathsInState, internalVersion)
+        ? _getPrunedExpandedPaths(expandedPathsInState, internalVersion)
         : getParentFolders(version.file.selected_file);
 
       return {
@@ -1498,7 +1499,7 @@ const reducer: Reducer<VersionsState, ActionType<typeof actions>> = (
       const version = getVersionInfo(state, versionId);
 
       if (version && expandedPaths) {
-        expandedPaths = getPrunedExpandedPaths(expandedPaths, version);
+        expandedPaths = _getPrunedExpandedPaths(expandedPaths, version);
       }
 
       return {
```

The stub can be a spy that returns the *real* `getPrunedExpandedPaths()`. This will make it so you don't have to do any special set up to simulate its return value. Example:

```js
reducer(
  undefined,
  actions.loadVersionInfo({ version: oldVersion }),
  { _getPrunedExpandedPaths: jest.fn(getPrunedExpandedPaths) }
)
```

The test can then verify that both `expandedPaths` and `internalVersion` are passed to `getPrunedExpandedPaths`. That's all it needs to check because this passes all other responsibility on to `getPrunedExpandedPaths()`, which has its own tests.